numpy/distutils/command/build_ext.py

```diff
@@ -262,15 +262,24 @@ def run(self):
         self.build_extensions()
 
         # Copy over any extra DLL files
-        runtime_lib_dir = os.path.join(
-            self.build_lib, self.distribution.get_name(), 'extra-dll')
-        for fn in os.listdir(self.extra_dll_dir):
-            if not fn.lower().endswith('.dll'):
-                continue
-            if not os.path.isdir(runtime_lib_dir):
-                os.makedirs(runtime_lib_dir)
-            runtime_lib = os.path.join(self.extra_dll_dir, fn)
-            copy_file(runtime_lib, runtime_lib_dir)
+        # FIXME: In the case where there are more than two packages,
+        # we blindly assume that both packages need all of the libraries,
+        # resulting in a larger wheel than is required. This should be fixed,
+        # but it's so rare that I won't bother to handle it.
+        pkg_roots = set(
+            self.get_ext_fullname(ext.name).split('.')[0]
+            for ext in self.extensions
+        )
+        for pkg_root in pkg_roots:
+            shared_lib_dir = os.path.join(self.build_lib, pkg_root,
+                                          '.libs')
+            if not os.path.isdir(shared_lib_dir):
```

Comment:
Note: if installing in editable mode, the `shared_lib_dir` should NOT be prefixed with `build_lib` if not using 2to3 (why are there so many different scenarios to handle?). Im just going to leave the latter scenario out for now and handle the former.